linera-views/src/memory.rs

```diff
@@ -135,13 +135,23 @@ impl KeyValueStore for MemoryStore {
 
 impl MemoryStore {
     /// constructor of MemoryStore
-    pub fn new(guard: MutexGuardArc<MemoryStoreMap>, max_stream_queries: usize) -> Self {
```

Comment:
The previous name `new` made more sense than `new_from_guard`. In general, I would keep `new` for methods that are the closest thing to a standard constructor (say, taking one value per field).